include/tsl/sparse_hash.h

```diff
@@ -1929,6 +1936,10 @@ class sparse_hash : private Allocator,
           m_nb_elements--;
           m_nb_deleted_buckets++;
 
+          if (size() + m_nb_deleted_buckets >= m_load_threshold_clear_deleted) {
+            clear_deleted_buckets();
```

Comment:
`clear_deleted_buckets` can throw an exception in some cases. A element could thus be correctly erased but the `erase` method could thus throw an exception which goes against the exception safety of the map.